### ISSUE TITLE
PHP Fatal error: Call to a member function getBody()

### ISSUE THREAD

**Issue by jonathaningram** [guzzle/guzzle#172]
I've just received the following error in my logs.

The exact error was (no track trace available):

```
PHP Fatal error: Call to a member function getBody() on a non-object in vendor/guzzle/http/Guzzle/Http/Curl/RequestMediator.php on line 100
```

The method of interest is:

``` php
<?php

    /**
     * Write data to the response body of a request
     *
     * @param resource $curl  Curl handle
     * @param string   $write Data that was received
     *
     * @return int
     */
    public function writeResponseBody($curl, $write)
    {
        if ($this->emitIo) {
            $this->request->dispatch('curl.callback.write', array(
                'request' => $this->request,
                'write'   => $write
            ));
        }

        return $this->request->getResponse()->getBody()->write($write); // line 100
    }
```

The error does not always happen, so is this a library error or a user error? Note: I'm using the Goutte library (https://github.com/fabpot/goutte/issues) which site on top of Guzzle, so it's possible the error comes from them.

If it helps, here is my Goutte code. I know it's Goutte code, but the fatal error comes from Guzzle.

``` php
<?php

    try {
            $proxy = $this->proxyProvider->provideAny(array(
                'disallowed_ports' => 80,
            ));

            if (null !== $proxy) {
                $this->client->getClient()->setConfig(array(
                    'curl.CURLOPT_PROXY'     => $proxy,
                    'curl.CURLOPT_PROXYTYPE' => 'CURLPROXY_HTTP'
                ));
            }

            $userAgentString = $this->userAgentGenerator->generate();
            $referrer = '';

            $this->client->setHeader('User-Agent', $userAgentString);
            $this->client->setHeader('Referer', $referrer);

            $this->client->request('GET', $url);
        } catch (CurlException $e) {
             // ...
        }

        $response = $this->client->getResponse();

        // ...
```

**Comment by thanhtuyen** [comment]
i goutte has function which the same curl_getinfo in php
Because I want to get all information on website below format
[url] => http://ione.vnexpress.net/sitemap.xml
    [content_type] => application/xml; charset=utf-8
    [http_code] => 200
    [header_size] => 220
    [request_size] => 68
    [filetime] => -1
    [ssl_verify_result] => 0
    [redirect_count] => 0
    [total_time] => 0.610405
    [namelookup_time] => 0.022581
    [connect_time] => 0.026514
    [pretransfer_time] => 0.026703
    [size_upload] => 0
    [size_download] => 562190
    [speed_download] => 921011
    [speed_upload] => 0
    [download_content_length] => -1
    [upload_content_length] => 0
    [starttransfer_time] => 0.516543
    [redirect_time] => 0
    [redirect_url] => 
    [primary_ip] => 192.65.248.157
    [certinfo] => Array
        (
        )

**Comment by mtdowling** [comment]
Can you rephrase the question?

**Comment by thanhtuyen** [comment]
I want to get infomation below format
[url] => http://ione.vnexpress.net/sitemap.xml
[content_type] => application/xml; charset=utf-8
[http_code] => 200
[header_size] => 220
[request_size] => 68
[filetime] => -1
[ssl_verify_result] => 0
[redirect_count] => 0
[total_time] => 0.610405
[namelookup_time] => 0.022581
[connect_time] => 0.026514
[pretransfer_time] => 0.026703
[size_upload] => 0
[size_download] => 562190
[speed_download] => 921011
[speed_upload] => 0
[download_content_length] => -1
[upload_content_length] => 0
[starttransfer_time] => 0.516543
[redirect_time] => 0
[redirect_url] => 
[primary_ip] => 192.65.248.157
[certinfo] => Array
(
)

But i don't know how to get it use Goute?
I want to see construct
